modules/manipulateGTC.py

```diff
@@ -73,31 +74,64 @@ def snpUpdate(data, line):
-        originalSnp = data[1003][loc]
-        data[1003][loc] = str(line.rstrip().split()[1]).encode()
-        if ((str(line.rstrip().split()[1])[0] != str(line.rstrip().split()[1])[1]) and (str(line.rstrip().split()[1])[0] != '-')):
+        manifest_name = manifest.names[loc]
+        manifest_name_csv = manifest_csv.names[loc]
+        assert manifest_name == manifest_name_csv
+
+
+        manifestSnpsStr = manifest.snps[loc].decode()
+        manifestSnps = [manifestSnpsStr[1], manifestSnpsStr[-2]]
+        if manifest.ref_strands[loc] == RefStrand.Minus:
+            manifestSnps = [COMPLEMENT_MAP[snp] for snp in manifestSnps]
+
+
+        newSnpsStr = str(line.rstrip().split()[1])
+        newSnps = [newSnpsStr[0], newSnpsStr[1]]
+
+        # Getting genotype byte
+        if newSnps[0] != newSnps[1] and newSnps[0] != '-':
             data[1002][loc] = 2
-        elif (str(line.rstrip().split()[1])[0] == '-') and (str(line.rstrip().split()[1])[1] == '-'):
+        elif newSnps[0] == '-' and newSnps[1] == '-':
             data[1002][loc] = 0
-        elif (str(line.rstrip().split()[1])[0] == str(line.rstrip().split()[1])[1]) and (str(line.rstrip().split()[1])[0] in ['A', 'T', 'G', 'C']) and (str(line.rstrip().split()[1])[1] in ['A', 'T', 'G', 'C']):
-            if manifest.snps[loc].decode().find(str(line.rstrip().split()[1])[0]) != -1:
-                data[1002][loc] = manifest.snps[loc].decode().find(str(line.rstrip().split()[1])[0])
+            data[1003][loc] = '--'.encode()
+        elif newSnps[0] == newSnps[1]:
+            if newSnps[0] in manifestSnps:
+                if newSnps[0] == manifestSnps[0]:
+                    data[1002][loc] = 1
+                else:
+                    data[1002][loc] = 3
             else:
-                logger.warning('WARNING! {} allele possibilities do not match manifest.  {}={} and manifest={}. This snp will not be updated.'
-                    .format(line.rstrip().split()[0],
-                            line.rstrip().split()[0], originalSnp,
-                            manifest.snps[loc]))
-                print(
-                    'WARNING! {} allele possibilities do not match manifest.  {}={} and manifest={}. This snp will not be updated.'
-                    .format(line.rstrip().split()[0],
-                            line.rstrip().split()[0], originalSnp,
-                            manifest.snps[loc]))
-                sys.stdout.flush()
-                data[1003][loc] = originalSnp
+                newSnps = [COMPLEMENT_MAP[snp] for snp in newSnps]
+                if newSnps[0] == manifestSnps[0]:
+                    data[1002][loc] = 1
+                else:
+                    data[1002][loc] = 3
         else:
-            pass
+            print("NO CONDITION MET; SOMETHING IS WRONG")
 
+        # Getting base call byte
+        if data[1002][loc] == 0:
+            data[1003][loc] = '--'.encode()
+        else:
+            if newSnps[0] in ['I', 'D']:
```

Comment:
YES!  This is the use case I never added or even thought of b/c at the time I don't think we cared about indel and were exclusively looking at SNPs so I never added the case nor did I ever have a use case that I tested it on, but this makes perfect sense.